src/WebAPI/ModelsAPI.cs

```diff
@@ -505,8 +505,23 @@ public static async Task<JObject> DoModelDownloadWS(Session session, WebSocket w
         }
         try
         {
-            string outPath = $"{handler.FolderPaths[0]}/{name}.safetensors";
-            if (File.Exists(outPath))
+            string baseModel = "";
+            if (!string.IsNullOrWhiteSpace(metadata))
+            {
+                JObject metadataObj = JObject.Parse(metadata);
+                baseModel = metadataObj["modelspec.baseModel"]?.ToString() ?? "";
+            }
+            string modelOutPath;
+            if (!string.IsNullOrWhiteSpace(baseModel) && session.User.Settings.GroupDownloadedModelsByBaseType)
+            {
+                modelOutPath = $"{handler.FolderPaths[0]}/{baseModel}/{name}.safetensors";
```

Comment:
Since this is just editing the folder path, it should be implemented in the `utiltab.js` rather than the C# here.
On top of avoiding cleaning/handling/API stability issues, that also gives the perk that you can just emit the folder prefix into the existing folder dropdown field, ie allowing users to edit the path on a case-by-case basis whenever they want

Reply:
How do I get access to user settings from the js side? On the C# side I could just reach for the user session pretty easily. I know the JS side must have a helper method for that as well, right?

Reply:
`src/wwwroot/js/genpage/settings_editor.js` `getUserSetting` maybe?

Reply:
yeah `getUserSetting`